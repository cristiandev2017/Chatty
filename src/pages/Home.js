import React, { Component } from 'react';
import Header from '../components/Header';
import Footer from '../components/Footer';
import { Link } from 'react-router-dom';

export default class HomePage extends Component {
  render() {
    return (
      <div className="home">
        <Header></Header>
        <section>
          <div className="jumbotron jumbotron-fluid py-5">
            <div className="container text-center py-5">
              <h1 className="display-4">Bienvenido a Chatty</h1>
              <p className="lead">Un lugar para compartir por medio de un chat con tus amigos</p>
              <div className="mt-4">
                <Link className="btn btn-primary px-5 mr-3" to="/signup">Crear Nueva Cuenta</Link>
                <Link className="btn px-5" to="/login">Login con tu Cuenta</Link>
              </div>
            </div>
          </div>
        </section>
        <Footer></Footer>
      </div>
    )
  }
}