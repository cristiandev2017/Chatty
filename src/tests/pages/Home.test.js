import { shallow } from "enzyme";
import Home from "../../pages/Home";
import React from "react";

describe("Pruebas del componente <Home/>", () => {
  //Arrange
  const wrapper = shallow(<Home />);

  test("Comprobar que corra el componente", () => {
    //Como es solo para revisar que funcione y el componente no tiene acciones
    //El Assert sera un snapshot
    expect(wrapper).toMatchSnapshot();
  });

  //Prueba del titulo
  test("Debe de mostrar el valor del titulo", () => {
    //Arrange
    const titulo = "Bienvenido a Chatty";
    //Act
    const textoTitulo = wrapper.find("h1").text();
    
    //Assert
    expect(textoTitulo).toBe(titulo);
  });

  //Prueba del texto
    //Puedo hacer pruebas de elementos
  test("Debe de mostrar el valor del texto", () => {
    //Arrange
    const texto = "Un lugar para compartir por medio de un chat con tus amigos";
    //Act
    const textoTexto = wrapper.find("p").text();
    
    //Assert
    expect(textoTexto).toBe(texto);
  });

});
