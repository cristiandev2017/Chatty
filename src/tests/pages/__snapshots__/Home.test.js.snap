// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <Home/> Comprobar que corra el componente 1`] = `
<div
  className="home"
>
  <Header />
  <section>
    <div
      className="jumbotron jumbotron-fluid py-5"
    >
      <div
        className="container text-center py-5"
      >
        <h1
          className="display-4"
        >
          Bienvenido a Chatty
        </h1>
        <p
          className="lead"
        >
          Un lugar para compartir por medio de un chat con tus amigos
        </p>
        <div
          className="mt-4"
        >
          <Link
            className="btn btn-primary px-5 mr-3"
            to="/signup"
          >
            Crear Nueva Cuenta
          </Link>
          <Link
            className="btn px-5"
            to="/login"
          >
            Login con tu Cuenta
          </Link>
        </div>
      </div>
    </div>
  </section>
  <Footer />
</div>
`;
